src/i18n-module.php

```diff
@@ -203,7 +202,7 @@ private function hide_promo() {
 	 * @return string The i18n promo message.
 	 */
 	public function get_promo_message() {
-		if ( ! $this->is_default_language( $this->locale ) ) {
+		if ( ( ! $this->is_default_language( $this->locale )  ) && ( ! $this->hide_promo() ) ) {
```

Comment:
`!` already binds stronger than `&&`, I would recommend removing the additional brackets for easier reading:

`if ( ! $this->is_default_language( $this->locale ) && ! $this->hide_promo() )`

Reply:
Thanks, didn't know that!